chanrestore.go

```diff
@@ -96,6 +96,18 @@ func (c *chanDBRestorer) openChannelShell(backup chanbackup.Single) (
 		return nil, fmt.Errorf("unknown Single version: %v", err)
 	}
 
+	// It is extremely unlikely at this point that a channel we are trying
+	// to restore has a coinbase funding TX. Therefore we can be quite
+	// certain that if the TxIndex is zero, it was an unconfirmed channel
+	// where we used the BlockHeight to encode the funding TX broadcast
+	// height. To not end up with an invalid short channel ID that looks
+	// valid, we restore the "original" unconfirmed one here.
+	fundingBroadcastHeight := uint32(0)
+	if backup.ShortChannelID.TxIndex == 0 {
+		fundingBroadcastHeight = backup.ShortChannelID.BlockHeight
+		backup.ShortChannelID.BlockHeight = 0
```

Comment:
We may want to consolidate this logic with https://github.com/lightningnetwork/lnd/pull/3690 to update the field in a single place.

Reply:
nice work @guggero, i think this will greatly limit the potential for any unintended side effects :)

Reply:
@wpaulino: right, makes sense. Closed #3690 and will add the changes to this PR.